aten/src/ATen/native/vulkan/api/Resource.h

```diff
@@ -27,22 +28,49 @@ struct Resource final {
       VkAccessFlags dst;
     };
 
+    /*
+      Descriptor
+    */
+
+    struct Descriptor final {
+      VmaMemoryUsage usage;
+      VkMemoryPropertyFlags /* optional */ required;
+      VkMemoryPropertyFlags /* optional */ preferred;
+    };
+
     VmaAllocator allocator;
     VmaAllocation allocation;
 
+    struct Access final {
+      typedef uint8_t Flags;
+
+      enum Type : Flags {
+        Read = 1u << 0u,
+        Write = 1u << 1u,
+      };
+
+      template<typename Type, Flags access>
+      using Pointer = std::add_pointer_t<
+          std::conditional_t<
+              0u != (access & Write),
```

Comment:
Could it be juts: `access & Write`?

Reply:
That's what I thought originally as well but unfortunately the compiler wasn't happy with that and kept giving me cryptic errors which took quite a while to root cause. :/  The first argument of std::conditional is a boolean and I am not quite sure why an implicit integer ((access & Write) returns an integer) to boolean conversion could not have been performed automatically in that context like everywhere else, so had to make this explicitly a boolean with a comparison against zero.